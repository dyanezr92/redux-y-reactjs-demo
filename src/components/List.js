import React from 'react'
import PropTypes from 'prop-types'
import InfoMessage from './InfoMessage'

const List = ({ elements, emptyMessage, children }) => {

  const emptyListWarning =
    elements.length === 0 && emptyMessage
      ? <InfoMessage text={ emptyMessage }/> : null;

  return (
    <div>
      { emptyListWarning }
      { children }
    </div>
  );
};

List.propTypes = {
  elements: PropTypes.array.isRequired,
  emptyMessage: PropTypes.string.isRequired
}

export default List
