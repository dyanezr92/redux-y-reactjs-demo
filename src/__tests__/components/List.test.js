import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'; // Smart components

import List from '../../components/List';

describe('<List elements={[]} emptyMessage="Empty!"/>', () => {
  describe('render()', () => {
    test('renders the list', () => {
      const wrapper = shallow(<List elements={[]} emptyMessage="Empty!"/>);
      expect(wrapper).toMatchSnapshot();
    });
  });
});

describe('<List elements={[a: "a", b: "b"]} emptyMessage="Empty!"/>', () => {
  describe('render()', () => {
    test('renders the list', () => {
      const wrapper = shallow(<List elements={["a": "a", "b": "b"]} emptyMessage="Empty!">Test children</List>);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
