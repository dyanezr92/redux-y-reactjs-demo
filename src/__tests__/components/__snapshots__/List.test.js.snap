// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List elements={[]} emptyMessage="Empty!"/> render() renders the list 1`] = `
<div>
  <InfoMessage
    text="Empty!"
  />
</div>
`;

exports[`<List elements={[a: "a", b: "b"]} emptyMessage="Empty!"/> render() renders the list 1`] = `
<div>
  Test children
</div>
`;
